packages/govern-lib/src/utils/NetworkContext.ts

```diff
@@ -0,0 +1,96 @@
+import { Address, Networkish, Network } from '../types'
+import { networkContexts } from '../config'
+import { ErrorInvalidNetwork } from '../errors'
+
+class NetworkContext implements Network {
+  readonly chainId: number
+  readonly ensAddress: Address
+  readonly name: string
+  readonly subgraphUrl: string
+
+  static chainIdFromName(name: string): number | undefined {
+    for (const [chainId, networkContext] of networkContexts.entries()) {
+      if (networkContext.name === name) {
+        return chainId
+      }
+    }
+    return undefined
+  }
+
+  static from(network: Networkish, subgraphUrl?: string): NetworkContext {
+    if (!network) {
+      throw new ErrorInvalidNetwork(`Network: incorrect value provided.`)
+    }
+
+    if (typeof network === 'string') {
+      return new NetworkContext({ name: network, subgraphUrl })
+    }
+
+    if (typeof network === 'number') {
+      return new NetworkContext({ chainId: network, subgraphUrl })
+    }
+
+    return new NetworkContext({ ...network, subgraphUrl })
+  }
+
+  constructor({
+    chainId,
+    ensAddress,
+    name,
+    subgraphUrl,
+  }: {
+    chainId?: number
+    ensAddress?: Address
+    name?: string
+    subgraphUrl?: string
+  }) {
```

Comment:
Totally not a fan of such constructors. I constructor itself should not have such logic. Please create methods for this.

Reply:
No strong opinion, but I find it easier to read when property assignments are done in the constructor, without having to jump to other methods. Is it what you are disagreeing with?

Reply:
Yeah, I just learned from OOP that no logic should be in the constructors. I think accessors are a great use case for something like this :-)